app/js/app/app.js

```diff
@@ -827,8 +829,14 @@ define([
         var checkForWebSocket = function() {
 
             if ($rootScope.notificationWebSocket != null) {
-                $rootScope.notificationWebSocket.send("user-snapshot-nudge");
-                $rootScope.webSocketCheckTimeout = $timeout(checkForWebSocket, 10000);
+                if (!$rootScope.user.userSnapshot) {
+                    // If we don't have a snapshot, request one.
+                    $rootScope.notificationWebSocket.send("user-snapshot-nudge");
+                } else {
+                    // Else just ping to keep connection alive.
+                    $rootScope.notificationWebSocket.send("heartbeat");
+                }
+                $rootScope.webSocketCheckTimeout = $timeout(checkForWebSocket, 60000);
```

Comment:
Just a quick note to point out that this would mean that different tabs could show inconsistent streak values for up to 60 seconds after midnight. We can live with that!